File: string_array/Majority_Element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        tracker = nums[0]
        counter = 0
        for i in range(0, len(nums)):
            if (nums[i] == tracker):
                counter += 1
            else:
                counter -= 1
                if (counter == 0):
                    tracker = nums[i]
                    counter = 1
        return tracker

File: string_array/Majority_Element.java
class Solution {
    public int majorityElement(int[] nums) {
        int tracker = nums[0];
        int counter = 0;
        for (int i = 0; i < nums.length; ++i) {
            if (nums[i] == tracker) {
                counter++;
            }
            else {
                counter--;
                if (counter == 0) {
                    tracker = nums[i];
                    counter = 1;
                }
            }
        }
        return tracker;
    }
}